source/base/partitioner.cc

```diff
@@ -325,6 +333,40 @@ namespace Utilities
 
 
 
+    bool
+    Partitioner::is_compatible (const Partitioner &part) const
+    {
+      // is the partitioner points to the same memory location as the calling
+      // processor
+      if (&part == this)
+        return true;
+#ifdef DEAL_II_WITH_MPI
+      if (Utilities::MPI::job_supports_mpi())
+        {
+          int communicators_same = 0;
+          MPI_Comm_compare (part.communicator, communicator,
+                            &communicators_same);
+          if (!(communicators_same == MPI_IDENT ||
+                communicators_same == MPI_CONGRUENT))
+            return false;
+        }
+#endif
+      return (global_size == part.global_size &&
+              local_range_data == part.local_range_data &&
+              ghost_indices_data == part.ghost_indices_data);
+    }
+
+
+
+    bool
+    Partitioner::is_globally_compatible (const Partitioner &part) const
+    {
+      return Utilities::MPI::min(static_cast<int>(is_compatible(part)),
+                                 communicator) == 1;
```

Comment:
what do you mean here? MPI_Comm_compare is in is_compatible (local version). It is meant for the case where one vector is default-constructed (MPI_COMM_SELF) and the other is a distributed vector that neither owns nor ghosts vector elements for a given processor. Without this check, some processors would see a compatible partitioner here whereas it clearly is incompatible.